babel/messages/frontend.py

```diff
@@ -1173,5 +1109,19 @@ def parse_keywords(strings: Iterable[str] = ()):
     return keywords
 
 
+try:
+    # Re-exports for backwards compatibility;
+    # `setuptools_frontend` is the canonical import location.
+    from babel.messages.setuptools_frontend import (
+        check_message_extractors,  # noqa: F401
+        compile_catalog,  # noqa: F401
+        extract_messages,  # noqa: F401
+        init_catalog,  # noqa: F401
+        update_catalog,  # noqa: F401
+    )
+except ImportError:
+    # We expect this to mean that neither setuptools or distutils are installed.
+    pass
```

Comment:
There's a great way (new in Python 3.7) to provide a deprecated member only on the fly!

If you want, you can even add a deprecation warning in the body. I skipped doing it for now because why not have a transition period without any deprecation message.

```suggestion
def __getattr__(name: str):
    # Re-exports for backwards compatibility;
    # `setuptools_frontend` is the canonical import location.
    if name in {'check_message_extractors', 'compile_catalog', 'extract_messages', 'init_catalog', 'update_catalog'}:
        from babel.messages import setuptools_frontend

        return getattr(setuptools_frontend, name)

    raise AttributeError(f"module {__name__!r} has no attribute {name!r}")

```

This is kind of a big deal also because setuptools is a relatively very slow import, great to be able to skip it if not necessary!

Reply:
Yeah, this is probably the better approach. I keep forgetting about module-level `__getattr__` 😁